src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -452,6 +453,60 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return type.GetUseSiteErrorInfo()
         End Function
 
+        Private Function GetInternalXmlHelperType(syntax As VisualBasicSyntaxNode, diagBag As DiagnosticBag) As NamedTypeSymbol
+            Debug.Assert(Me.ContainingType IsNot Nothing)
+            Dim containingAssembly = Me.ContainingType.ContainingAssembly
+
+            Dim typeSymbol = GetInternalXmlHelperType(containingAssembly)
+
+            Dim useSiteError = GetUseSiteErrorForWellKnownType(typeSymbol)
+            If useSiteError IsNot Nothing Then
+                ReportDiagnostic(diagBag, syntax, useSiteError)
+            End If
+
+            Return typeSymbol
+        End Function
+
+        Private Shared Function GetInternalXmlHelperType(containingAssembly As AssemblySymbol) As NamedTypeSymbol
+            Const metadataName = "My.InternalXmlHelper"
+
+            Dim typeSymbol = containingAssembly.GetTypeByMetadataName(
+                metadataName,
+                includeReferences:=True,
+                isWellKnownType:=True)  ' It's not, but we want the extra validation.
+
+            If typeSymbol Is Nothing Then
+                Dim emittedName = MetadataTypeName.FromFullName(metadataName)
+                typeSymbol = New MissingMetadataTypeSymbol.TopLevel(containingAssembly.Modules(0), emittedName)
```

Comment:
I changed it to create a MissingMetadataTypeSymbol in the module we were searching.